### PR TITLE
print all versions of tendermint and its sub protocols 

### PR DESCRIPTION
**Pull request by marbar3778**
print all versions of tendermint not only the main one

could we get this into 0.34 as well?

---

#### PR checklist

- [ ] Tests written/updated, or no tests needed
- [ ] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [ ] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: cmd/tendermint/commands/version.go [hunk 1]
```diff
@@ -11,8 +12,20 @@ import (
...
-	Short: "Show version info",
+	Short: "Show protocols' and libraries' version numbers",
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println(version.TMCoreSemVer)
+		values, _ := json.MarshalIndent(struct {
+			Tendermint    string
+			ABCI          string
+			BlockProtocol uint64
+			P2PProtocol   uint64
+		}{
```
Comment (thanethomson) [inline cmd/tendermint/commands/version.go:22]: Is the idea with the JSON format that this is going to be machine-interpreted? Who or what would be interpreting it?

And why the difference in JSON field naming convention as compared to what's used, for example, in the genesis file (`Tendermint` as opposed to `tendermint`, and `BlockProtocol` vs `block_protocol`)?

Comment (marbar3778) [reply cmd/tendermint/commands/version.go:22]: did json format cause otherwise it looks ugly when printed. I can Rename if needed, I see this way as nicer

Comment (thanethomson) [reply cmd/tendermint/commands/version.go:22]: I have a few suggestions here:
1. Let's please rather keep the current `version` command default behaviour as-is (just printing the Tendermint version). I don't know if we _always_ need to see all versions of all protocols in that version.
2. Could we add a `-v`/`--verbose` flag to the `version` command to print this JSON struct?
3. Let's please stick to the current JSON conventions used throughout the codebase, which favour `snake_case`.

Comment (cmwaters) [reply cmd/tendermint/commands/version.go]: Made the respective updates. 

I don't think this requires a changelog as it doesn't really change the behaviour in a notable way but let me know if you disagree

**Review by cmwaters — APPROVED** [review]

**Pull request merged**